server/api/impl/aws/provision_app.go

```diff
@@ -146,10 +176,55 @@ func (c *Client) provisionApp(appID apps.AppID, appVersion apps.AppVersion, func
 		if err != nil {
 			return errors.Wrap(err, "can't get function name")
 		}
-		if err := c.createFunction(function.zipFile, name, function.handler, function.runtime, policyName); err != nil {
-			return errors.Wrapf(err, "can't install function for %s", appID)
+
+		if shouldUpdate {
+			if err := c.createOrUpdateFunction(function.zipFile, name, function.handler, function.runtime, policyName); err != nil {
+				return errors.Wrapf(err, "can't install function for %s", appID)
+			}
+		} else {
+			if err := c.createFunction(function.zipFile, name, function.handler, function.runtime, policyName); err != nil {
+				return errors.Wrapf(err, "can't install function for %s", appID)
+			}
 		}
 	}
+
+	return nil
+}
+
+func (c *Client) createOrUpdateFunction(zipFile io.Reader, function, handler, runtime, resource string) error {
```

Comment:
Why would we ever update a function? Dev-mode?

Reply:
Yes, updating a function should only happen in dev-mode. Never in production.